pkgs/top-level/aliases.nix

```diff
@@ -1482,6 +1482,9 @@ mapAliases ({
   swift-im = throw "swift-im has been removed as it is unmaintained and depends on deprecated Python 2 / Qt WebKit"; # Added 2023-01-06
   swig1 = throw "swig1 has been removed as it is obsolete"; # Added 2024-08-23
   swig2 = throw "swig2 has been removed as it is obsolete"; # Added 2024-08-23
+  swig3 = throw "swig3 has been removed as it is obsolete"; # Added 2024-09-12
+  swig4 = swig; # Added 2024-09-12
```

Comment:
A bit curious about the rationale for making `swig4` an alias rather than leaving it as a definition in `all-packages.nix`?

Reply:
My hope is that we can stop carrying multiple versions, so this is to forbid pinning the current version in‐tree, with a view towards dropping the alias. If you think that’s too ambitious then I can put it back (though the convention would be `swig_4`, which is another reason I felt like we should just drop it). So far it’s seemed like the number of things that break from new versions is relatively limited, and no one package has seemed like it has taken an inordinate amount of effort to fix during a normal `staging` cycle.

Reply:
I think it should be fine given recent experience? (Just wanted to confirm the intent was in fact to forbid pinning.)

Reply:
Yeah, my thinking is that we can always move it back if SWIG 5 somehow breaks everything, but that at present nobody really has any way of knowing that’s likely so pinning is a bad idea. (Same reason I unpinned all the FFmpeg 7 users.)